packages/@aws-cdk/aws-appsync/lib/schema-base.ts

```diff
@@ -135,7 +137,14 @@ export interface IIntermediateType {
    *
    * @default - no intermediate type
    */
-  readonly intermediateType?: InterfaceType;
+  readonly intermediateType?: IIntermediateType;
+
+  /**
+   * Bind this Intermediate Type to a GraphQL Api
+   *
+   * @param api The binding GraphQL Api [disable-awslint:ref-via-interface]
+   */
+  bindToGraphqlApi(api: GraphqlApi): IIntermediateType;
```

Comment:
our `bind` methods typically take in a scope (even if it's not used), and a struct of `Options` (even if it's empty) as well. It's more future friendly. 

They also return a `xyzConfig` object with the properties that are to be bound. This is a slightly different approach.